architecture.md

```diff
@@ -0,0 +1,34 @@
+RedditClone
+
+Users
+Posts
+Comments
+
+Users create posts, comments.  Users have total number of upvotes associated w/ account.
+Posts have comments from different users.  Posts are created by users and have upvote score as well.
+Comments can have replies from other comments.  Comments are created by users and also have upvotes.
+
+
+MongoDB Schema
+
+User - id
+Username - string
+postHistory - List of Posts
+commentHistory - List of Comments
+postUpvotes - int (total upvotes from posts created by user)
+commentUpvotes - int (total upvotes from comments created by user)

```

Comment:
you plan to keep these in sync with the upvotes from posts/comments rather than just calculate them on the fly when they are needed?